compiler/lib/partial_cps_analysis.ml

```diff
@@ -96,15 +96,6 @@ let block_deps ~info ~vars ~tail_deps ~deps ~blocks ~fun_name pc =
               (* If a function contains effect primitives, it must be
                  in CPS *)
               add_dep deps f x)
-      | Let (x, Prim (Extern "caml_assume_no_perform", _)) -> (
-          add_var vars x;
-          match fun_name with
-          | None -> ()
-          | Some f ->
-              add_var vars f;
-              (* If a function contains effect primitives, it must be
-                 in CPS *)
-              add_dep deps f x)
```

Comment:
Couldn’t we exploit the fact that functions called under assume_no_perform may not need to be in CPS?

Reply:
Not easily. At the moment, we are considering that all escaping functions need to be in CPS. But I'm planning to change that.

Reply:
We are not marking anywhere explicitly the function passed to `caml_assume_no_perform` as needing CPS.
Any value passed to a primitive is considered as escaping by the data flow analysis. And any escaping function is marked as needing CPS in function `cps_needed`.

Reply:
Ah, I see.